hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.node.app.service.mono.state.virtual.EntityNumValue;
+import com.hedera.node.app.service.mono.state.virtual.EntityNumVirtualKey;
+import com.hedera.node.app.spi.state.WritableKVState;
+import com.hedera.node.app.spi.state.WritableKVStateBase;
+import com.hedera.node.app.spi.state.WritableStates;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Provides write methods for modifying underlying data storage mechanisms for working with
+ * accounts.
+ *
+ * <p>This class is not exported from the module. It is an internal implementation detail. This
+ * class is not complete, it will be extended with other methods like remove, update etc.,
+ */
+public class WritableAccountStore extends ReadableAccountStore {
+    /** The underlying data storage class that holds the account data. */
+    private final WritableKVState<EntityNumVirtualKey, Account> accountState;
```

Comment:
There are some pros and cons, but I'd lean toward not using any `mono-service` K/V types like `EntityNumVirtualKey` in `*-service-impl` code.

But this may have implications for existing `*ParityTest` unit tests, so we should think through separately from this PR.

Reply:
Will replace to use `EntityNum`

Reply:
Seems like it changes lot of files. Will do in next PR 😄